devserver.sh

```diff
@@ -1,23 +1,39 @@
 #!/bin/sh
 
-work_dir=$(mktemp -d)
+work_dir="$(mktemp -d)"
 
-if [ -z $MVN ]; then
-    MVN=$(which mvn)
+function cleanup() {
+    podman stop cryostat-devserver-reports
+    rm -rf "${work_dir}"
+}
+trap cleanup EXIT
+
+if [ -z "${MVN}" ]; then
+    MVN="$(which mvn)"
 fi
 
 for i in archive clientlib conf templates probes; do
-    if [ -e $work_dir/$i ]; then
-        if [ ! -d $work_dir/$i ]; then
-            echo "$work_dir/$i already exists but is not a directory"
+    if [ -e "${work_dir}/${i}" ]; then
+        if [ ! -d "${work_dir}/${i}" ]; then
+            echo "${work_dir}/${i} already exists but is not a directory"
             exit 1
         fi
     else
-        mkdir $work_dir/$i
+        mkdir "${work_dir}/${i}"
     fi
 done
 
+if [ -z "$CRYOSTAT_CORS_ORIGIN" ]; then
+    "${MVN}" -DskipTests=true prepare-package
+else
+    "${MVN}" -DskipTests=true -Dcryostat.minimal=true prepare-package
+fi
```

Comment:
What exactly is this `if-else` block doing?

Reply:
> However, in the original dev-server PR we were able to launch a hot-reloading instance of the web-client without involving the CRYOSTAT_CORS_ORIGIN flag

https://github.com/cryostatio/cryostat/pull/809/files#diff-ad8ed4ad8729e9fd13d92c8775318283cf8f2b58c1a49edd9eb3a0c195257cd1L35

Previously, the devserver script just always had the CORS origin environment variable set to the "standard" value.

> What exactly is this if-else block doing?

If you're using a separate web-client devserver then you must set the CORS origin to allow it to make requests to the backend. If that's the case then there is no need for the backend devserver to also build the web-client frontend assets. If you are not using a frontend devserver and have no CORS setup, then it does build the web-client frontend assets so that it can serve them to you.

Reply:
`prepare-package` does everything required for `package`, except it doesn't do the final step of actually creating the OCI image. Classes are generated, classes are compiled, resources are built/generated and copied into the correct locations, etc. This basically populates the `target/classes` and `target/assets` directories. With `-Dcryostat.minimal=true` the front-end build portion is skipped, so `target/assets` won't contain the HTML/JS/CSS of -web.

Reply:
Makes sense, thanks.